battles/forms.py

```diff
@@ -0,0 +1,123 @@
+from django import forms
+
+from battles.models import Battle, TrainerTeam
+from pokemons.helpers import get_pokemon_stats, is_pokemons_sum_valid
+from pokemons.models import Pokemon
+
+
+class CreateBattleForm(forms.ModelForm):
+    class Meta:
+        model = Battle
+        fields = ('trainer_opponent', 'trainer_creator', 'status')
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields['trainer_creator'].initial = self.initial['user'].id
+        self.fields['trainer_creator'].widget = forms.HiddenInput()
+        self.fields['status'].initial = 'ON_GOING'
+        self.fields['status'].widget = forms.HiddenInput()
+
+        pkm1 = forms.CharField()
+        pkm2 = forms.CharField()
+        pkm3 = forms.CharField()
+        self.fields['pokemon_1'] = pkm1
+        self.fields['pokemon_2'] = pkm2
+        self.fields['pokemon_3'] = pkm3
```

Comment:
you could declare these fields' widgets in Meta.widgets attribute instead of the constructor, I think it would be easier to read.

And about the initial values it would be better not to override them indefinitely, but only when no initial is passed on the form initialization. Something like that:

```python
pokemon_1 = forms.CharField()
pokemon_2 = forms.CharField()
pokemon_3 = forms.CharField()

class Meta:
    model = Battle
    fields = ('trainer_opponent', 'trainer_creator', 'status')
    widgets = {
        "trainer_creator": forms.HiddenInput(),
        "status": forms.HiddenInput(),
    },

def __init__(self, *args, **kwargs):
    initial = kwargs.get('initial', {})
    initial_user_id = initial.get('user') and initial.get('user').id 
    initial['trainer_creator'] = initial.get('trainer_creator', initial_user_id)
    initial['status'] = initial.get('status', 'ON_GOING')
    kwargs['initial'] = initial
    super().__init__(*args, **kwargs)

Reply:
I didn't know I could do this way. Thank you!